lib/rihanna/job.ex

```diff
@@ -177,6 +177,65 @@ defmodule Rihanna.Job do
   end
 
   @doc false
+  def delete_by(opts) do
+    ids_to_delete =
+      opts
+      |> filter_term_list()
+      |> Enum.reduce("", fn [id, _term], acc ->
+        "#{acc}'#{id}',"
+      end)
+      |> String.replace_trailing(",", "")
```

Comment:
It's not clear what this first block does. Can it be pulled into another function with a descriptive name?

Reply:
It's also rather inefficient due to the repeated binary concatenation. I think it's doing Enum.join here

Reply:
ugh, I just made the old mistake of re-inventing an Enum function using reduce